### PR TITLE
CB-7356 Adding new Ranger cloud paths for data and logs.

### PR DESCRIPTION
**Pull request by orngejaket**
This adds in 2 new cloud storage paths for Ranger for data and logs for CM > 7.2.1 and only for ALDSv2.

### CODE REVIEW

File: template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/ranger/RangerCloudStorageServiceConfigProviderTest.java [hunk 1]
```diff
@@ -34,19 +40,54 @@
...
+        TemplatePreparationObject preparationObject = getTemplatePreparationObjectForAWS(true)
                 .withBlueprintView(mock(BlueprintView.class))
+                .withProductDetails(new ClouderaManagerRepo().withVersion("7.2.1"), List.of())
                 .build();
 
         List<ApiClusterTemplateConfig> serviceConfigs = underTest.getServiceConfigs(templateProcessor, preparationObject);
 
         assertEquals(1, serviceConfigs.size());
-        assertEquals("ranger_plugin_hdfs_audit_url", serviceConfigs.get(0).getName());
+        assertEquals(RANGER_HDFS_AUDIT_URL, serviceConfigs.get(0).getName());
```
Comment (kmanamcheri) [inline template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/ranger/RangerCloudStorageServiceConfigProviderTest.java:53]: This is not recommended. Your tests should not be importing these constants. The reason is that if you do this, then if someone makes a typo in "ranger_plugin_hdfs_audit_url", you won't be catching it. 

By making this constant explicit in the tests, a developer has to make a conscious decision on making the change for constants.

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/ranger/RangerCloudStorageServiceConfigProvider.java [hunk 2]
```diff
@@ -18,13 +21,28 @@
 @Component
 public class RangerCloudStorageServiceConfigProvider implements CmTemplateComponentConfigProvider {
 
-    private static final String RANGER_HDFS_AUDIT_URL = "ranger_plugin_hdfs_audit_url";
+    static final String RANGER_HDFS_AUDIT_URL = "ranger_plugin_hdfs_audit_url";
```
Comment (kmanamcheri) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/ranger/RangerCloudStorageServiceConfigProvider.java:24]: These should be private. See my comment about using constants in tests.

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/ranger/RangerCloudStorageServiceConfigProvider.java [hunk 3]
```diff
@@ -18,13 +21,28 @@
...
+    static final String RANGER_LOG_URL = "cloud_log_location_url";
 
     @Override
     public List<ApiClusterTemplateConfig> getServiceConfigs(CmTemplateProcessor templateProcessor, TemplatePreparationObject templatePreparationObject) {
-        return ConfigUtils.getStorageLocationForServiceProperty(templatePreparationObject, RANGER_HDFS_AUDIT_URL)
-                .map(location -> List.of(config(RANGER_HDFS_AUDIT_URL, location.getValue())))
-                .orElseGet(() -> List.of(config(RANGER_HDFS_AUDIT_URL, getDefaultRangerAuditUrl(templateProcessor, templatePreparationObject))));
+        List<ApiClusterTemplateConfig> rangerConfigs = new ArrayList<>();
+
+        ConfigUtils.getStorageLocationForServiceProperty(templatePreparationObject, RANGER_HDFS_AUDIT_URL)
```
Comment (kmanamcheri) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/ranger/RangerCloudStorageServiceConfigProvider.java:34]: nit: This whole thing is hard to parse. Consider rewriting with extracting some of the variables.
```
        List<ApiClusterTemplateConfig> rangerConfigs = new ArrayList<>();

        final String rangerHdfsAuditUrl = ConfigUtils.getStorageLocationForServiceProperty(templatePreparationObject, RANGER_HDFS_AUDIT_URL)
            .map(location -> location.getValue())
            .orElseGet(() -> getDefaultRangerAuditUrl(templateProcessor, templatePreparationObject));

       rangerConfigs.add(config(RANGER_HDFS_AUDIT_URL, rangerHdfsAuditUrl));

       final String rangerDataUrl = ...
       final String rangerLogUrl = ...
       rangerConfigs.add(...);
       rangerConfigs.add(...);

      return rangerConfigs;
```

File: template-manager-cmtemplate/src/main/resources/definitions/cm-cloud-storage-location-specification.json [hunk 4]
```diff
@@ -98,6 +98,28 @@
...
       "requiredForAttachedCluster": true,
       "supportedStorages": ["ADLS", "WASB", "S3", "GCS", "ADLS_GEN_2"]
+    },
+    {
+      "type": "RANGER_DATA",
+      "propertyName": "cloud_data_location_url",
+      "propertyFile": "ranger-audit",
+      "description": "This is the path where Ranger stores its data",
+      "defaultPath": "{{{ defaultPath }}}/ranger/data",
+      "propertyDisplayName": "Ranger Cloud Data Location URL",
```
Comment (kmanamcheri) [inline template-manager-cmtemplate/src/main/resources/definitions/cm-cloud-storage-location-specification.json:108]: Should we display something related to Raz here?

Comment (orngejaket) [reply template-manager-cmtemplate/src/main/resources/definitions/cm-cloud-storage-location-specification.json:108]: Not technically Raz stuff.  This is all on Ranger.

File: template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/ranger/RangerCloudStorageServiceConfigProviderTest.java (hunk 1 above)
Comment (orngejaket) [reply template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/ranger/RangerCloudStorageServiceConfigProviderTest.java:53]: That's probably not needed, these fields are not configurable.

Comment (kmanamcheri) [reply template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/ranger/RangerCloudStorageServiceConfigProviderTest.java:53]: As a general principle, your tests shouldn't use imported constants. Yes they are not configurable but a dev can accidentally cause a regression by accidentally changing the constant.

It has happened before :)
